### PR TITLE
Salto-2615: add option to make recurseInto values not to fail the instance - Jira

### PR DESCRIPTION
**Pull request by shayc331**
_Add a flag to `RecurseIntoConfig` which won't fail the instance when we fail to retrieve some property from the service and use it for the `contextDefaults` field under `Field` instances_

---

_In some cases we want to allow skipping values where it might happen in order to avoid unintentionally omitting the instances_

---
_Release Notes_: 
_Jira_:
* Fixed issue where Field instances were omitted if their default value could not be fetched (e.g. Category)
---
_User Notifications_: 
_Jira_:
* Some Field instances that were erroneously omitted due to failure to get some of their attributes will be added (e.g. Category)

### CODE REVIEW

File: packages/adapter-components/src/elements/swagger/instance_elements.ts [hunk 1]
```diff
@@ -379,32 +380,39 @@ const getEntriesForType = async (
...
+          if (nested.isSingle) {
+            if (nestedEntries.length === 1) {
+              return [nested.toField, nestedEntries[0]] as [string, Values]
+            }
+            log.warn(`Expected a single value in recurseInto result for ${typeName}.${nested.toField} but received: ${nestedEntries.length}, keeping as list`)
+          }
+          return [nested.toField, nestedEntries] as [string, Values[]]
+        } catch (error) {
+          log.warn(`Failed getting extra field values for ${typeName} entry: ${safeJsonStringify(entry)}. Error: ${error.message}`)
+          if (nested.skipOnError) {
```
Comment (tamtamirr) [inline packages/adapter-components/src/elements/swagger/instance_elements.ts:401]: Can we please add a unit test for this scenario?

File: packages/adapter-components/src/elements/swagger/instance_elements.ts [hunk 2]
```diff
@@ -379,32 +380,39 @@ const getEntriesForType = async (
...
+          })
+          if (nested.isSingle) {
+            if (nestedEntries.length === 1) {
+              return [nested.toField, nestedEntries[0]] as [string, Values]
+            }
+            log.warn(`Expected a single value in recurseInto result for ${typeName}.${nested.toField} but received: ${nestedEntries.length}, keeping as list`)
+          }
+          return [nested.toField, nestedEntries] as [string, Values[]]
+        } catch (error) {
+          log.warn(`Failed getting extra field values for ${typeName} entry: ${safeJsonStringify(entry)}. Error: ${error.message}`)
```
Comment (netama) [inline packages/adapter-components/src/elements/swagger/instance_elements.ts:400]: we can make the logs more informative now - 
1. we can provide information about the specific nested field that is failing (`nested.toField`)
2.  let's have different log levels and messages based on whether `skipOnError` is true - 
* when it is true, the log level should be `info` and should indicate that only this specific field is omitted
* when it is false, the log level should be `warn` like it is today, and should indicate that the entire instance will be omitted

can be something like
```
          log.info(`Failed getting extra field values for field ${entry.toField} in ${typeName} entry: ${safeJsonStringify(entry), and the field will be omitted. Error: ${error.message}`)
```
and
```
          log.warn(`Failed getting extra field values for field ${entry.toField} in ${typeName} entry: ${safeJsonStringify(entry)}, not creating instance. Error: ${error.message}`)
```

File: packages/adapter-components/src/elements/swagger/instance_elements.ts [hunk 3]
```diff
@@ -379,32 +380,39 @@ const getEntriesForType = async (
...
+          if (nested.skipOnError) {
+            return undefined
           }
-          log.warn(`Expected a single value in recurseInto result for ${typeName}.${nested.toField} but received: ${nestedEntries.length}, keeping as list`)
+          throw error
         }
```
Comment (netama) [inline packages/adapter-components/src/elements/swagger/instance_elements.ts:405]: nit: if you add `.filter(isDefined)` here, you won't have to change the function signature (we don't get any value from returning `undefined` without any context)

**Review by netama — COMMENTED** [review]
looks good :) added a couple of comments. also:
1. as @tamtamirr mentioned, please add tests - you can use [this test](https://github.com/salto-io/salto/blob/main/packages/adapter-components/test/elements/swagger/instance_elements.test.ts#L1201) as a reference
2. the release notes and user notifications should be a little more specific - maybe something like:

* Release notes:
> Fixed issue where Field instances were omitted if their default value could not be fetched (e.g. `Category`)

* User notifications:
> Some Field instances that were erroneously omitted due to failure to get some of their attributes will be added (e.g. `Category`)

**Pull request closed without merging**